### ISSUE TITLE
Thread Panic from `grenad` - `must be greater than`

### ISSUE THREAD

**Issue by ryanrussell** [meilisearch/meilisearch#3043]
**Describe the bug**
During ingest, and trying to re-create the panic from #3042, I also saw this:

```bash
thread '<unnamed>' panicked at '[3, 115, 49, 55, 48, 49, 0, 50] must be greater than [3, 115, 49, 55, 48, 49, 0, 51]', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/grenad-0.4.3/src/block_writer.rs:109:17

thread '<unnamed>' panicked at '[3, 115, 49, 55, 48, 49, 0, 50] must be greater than [3, 115, 49, 55, 48, 49, 0, 53]', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/grenad-0.4.3/src/block_writer.rs:109:17
```

**To Reproduce**
Have not found minimal, reproducible example. 

Believe it is related to changing the ETL to run in multiple workers who may re-write the same indices/ID's.

I have moved ETL scripts into Airflow and parallelized them, so I am *GUESSING* there are instances where inserts from different workers are over-writing the same ID in close succession.

**Expected behavior**
Handle parallel ETL inserts dealing with the same ID without crash

**Screenshots**

**Meilisearch version:** v0.30.0rc1

**Additional context**
This *may* be related to #3042 -- It was seen during the same ETL trials.

---

TODO
- [x] Implement changes in [Milli](https://github.com/meilisearch/milli/)
- [ ] Release a Milli version containing these changes
- [ ] Bump this new Milli version in Meilisearch and merge it into `release-v0.30.0`

**Comment by curquiza** [comment]
Re opening, not fixed until we merged the new milli's version into `release-v0.30.0`

**Comment by curquiza** [comment]
Closed by https://github.com/meilisearch/meilisearch/pull/3084
